diff --git a/01-race-detector/demo/counter.go b/01-race-detector/demo/counter.go
index 6218487..0014adb 100644
--- a/01-race-detector/demo/counter.go
+++ b/01-race-detector/demo/counter.go
@@ -14,11 +14,10 @@ type incrementor struct {
 }
 
 func (inc *incrementor) increment() {
-	inc.mu.Lock()
-	defer inc.mu.Unlock()
-
 	for range 100000 {
+		inc.mu.Lock()
 		inc.counter++
+		inc.mu.Unlock()
 	}
 }
 
