diff --git a/01-race-detector/demo/counter.go b/01-race-detector/demo/counter.go
index a9c6eaa..6218487 100644
--- a/01-race-detector/demo/counter.go
+++ b/01-race-detector/demo/counter.go
@@ -2,6 +2,9 @@ package main
 
 import (
 	"fmt"
+	"log"
+	"os"
+	"runtime/trace"
 	"sync"
 )
 
@@ -20,6 +23,15 @@ func (inc *incrementor) increment() {
 }
 
 func main() {
+	f, err := os.Create("counter.trace")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer f.Close()
+
+	trace.Start(f)
+	defer trace.Stop()
+
 	var wg sync.WaitGroup
 	inc := &incrementor{}
 
